### PR TITLE
feat(client):add e2e coverage

### PR DESCRIPTION
**Pull request by Ismailtlem**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Related to https://github.com/freeCodeCamp/chapter/issues/786

- Added the packages `@cypress/code-coverage` for code coverage and `babel-plugin-istanbul` for instrumenting the code. The code coverage report will be in the folder cypress-coverage

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
Hey @Ismailtlem let me know when you'd like me to review this.

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "stop:docker": "docker-compose down",
     "cypress:open": "cypress open",
     "cypress:run": "cypress run",
+    "cypress:coverage:report": "npx nyc report --reporter=text-summary",
```
Comment (ojeytonwilliams) [inline package.json:32]:
```suggestion
    "cypress:coverage:report": "npx nyc report",
```
Could we try this?

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
This is great, thanks @Ismailtlem!

One thing I was wondering: how does the full report look in CI?  It would be nice to see the details, so we know what needs to be improved:

File: package.json (hunk 1 above)
Comment (Ismailtlem) [reply package.json]: Resolved

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
This is great, I really like the expanded output.

**Pull request merged**